### DISCUSSION TITLE
`out:fade` feels unpredictable when navigating pages on iOS Safari

### DISCUSSION THREAD

**Discussion by mostrecent** [sveltejs/kit#4856] [Q&A]
This is a very nuanced problem: You go back from `/some-page` to `/` by pressing the back button on iOS Safari (not by swiping to the right, that's a different case for now)—two options could happen:

_btw, `/some-page` has `out:fade` in its top `<div />` and `/` no fade at all for keep this example simple_

1. `/some-page` fades out and _then_ you see `/`
2.  `/some-page` is fading out  and `/` pops in at the same time

When doing this in SvelteKit's demo app I get option 1 and in my local app I get 2. Don't ask me why, I am testing my code for days now and can't find a clue when what happens.

Interestingly, both do not work when using the swipe gesture when going back. Hard to explain: You see `/`, for a split second, then `/some-page` flickering in with its fade out motion and then again `/`. This is a bigger issue which I think can't be solved and due to iOS Safari's slide back animation.

How do you guys handle this, better not use `out:fade` for page transitions at all or do I miss something?

**Comment by rowantrollope** [comment]
I am also seeing the page flicker when swiping back & forward.  Doesn't seem to happen on all pages, and doesn't happen when using the back and forward buttons.  same thing in chrome on ios when swiping.  Both are OK when clicking back and forward

**Comment by bbtgnn** [comment]
Same issue here in 2026
